### ISSUE TITLE
Add a tab Component  

### ISSUE THREAD

**Issue by Texhnolyze47** [fulldotdev/ui#80]
### Describe the feature

A tab component like nextui or or shadcn

**Comment by silveltman** [comment]
Noted, thanks for the request @Texhnolyze47 !

**Comment by vaagmay** [comment]
Hello @silveltman ,

Is there any update coming in Fulldev UI? 

Because the "preview/code" functionality is removed from the blocks. 
![fulldev ui](https://github.com/user-attachments/assets/8a4ee4a7-1399-4870-8458-f2b2309a1a5e)

**Comment by silveltman** [comment]
The fact that the button was missing was actually a bug, so thanks for letting know! Got that fixed now in 0.4.30.

That said, a tab component is not our priority at the moment. We're currently working on the v0.5 beta in which the priority is Layouts, Schemas and content collections/layers.

Basically, fulldev-ui is going in a similar direction as Starlight, but more focussed on building content/marketing websites, rather than documentation sites.

After we've reached a stable foundation we will focus on adding more components/blocks, like a Tab.

**Comment by vaagmay** [comment]
Ok @silveltman I got that button. Thanks for fixing and replying.

Is it possible to apply a component like progress bar ( i.e. that tracking the progress or to track the user where he left off from the site) in the content/marketing or in documentation websites.

If yes then please add this to your new version. Because I'm developing a documentation website and my content is too large so it is very hard to find where I left off. And in future user is also face the same. So I need help on that.

And when your new version is coming???

**Comment by silveltman** [comment]
@vaagmay I'm not sure what you mean by progress bar, do you mean a bar to indicate where on the page you're scrolled. This would probably not be something we will include in next release. 

We're planning on releasing the next version end of this week, might be as a beta (which is actually already on NPM but docs is not ready yet)

**Comment by SalahAdDin** [comment]
> [@vaagmay](https://github.com/vaagmay) I'm not sure what you mean by progress bar, do you mean a bar to indicate where on the page you're scrolled. This would probably not be something we will include in next release.
> 
> We're planning on releasing the next version end of this week, might be as a beta (which is actually already on NPM but docs is not ready yet)

Open a new issue for the progress bar. please!

**Comment by silveltman** [comment]
Closing due to the fact that we will use shadcn tab in the near future

**Comment by silveltman** [comment]
@SalahAdDin yes, the next version will use react + shadcn. You can have a look at branch `blocks-react`. 

This repo will become an extended version of shadcn, so extra components + blocks that are compatible with it.

Important to know is that it will still focus on usage within Astro, as this library is meant for content-drives sites for which I believe Astro is still the best framework, even though we use .tsx components instead of .astro.

**Comment by SalahAdDin** [comment]
> [@SalahAdDin](https://github.com/SalahAdDin) yes, the next version will use react + shadcn. You can have a look at branch `blocks-react`.
> 
> This repo will become an extended version of shadcn, so extra components + blocks that are compatible with it.
> 
> Important to know is that it will still focus on usage within Astro, as this library is meant for content-drives sites for which I believe Astro is still the best framework, even though we use .tsx components instead of .astro.

The good about this library was having vanilla JS components, It changed.

So bad!
